riscv/csrs.cc

```diff
@@ -467,7 +467,10 @@ mstatus_csr_t::mstatus_csr_t(processor_t* const proc, const reg_t addr):
       | (proc->extension_enabled_const('S') && (proc->get_const_xlen() != 32) ? set_field((reg_t)0, MSTATUS_SXL, xlen_to_uxl(proc->get_const_xlen())) : 0)
 
 #ifdef RISCV_ENABLE_DUAL_ENDIAN
-      | (proc->get_mmu()->is_target_big_endian() ? MSTATUS_UBE | MSTATUS_SBE | MSTATUS_MBE : 0)
+      | (proc->get_mmu()->is_target_big_endian() ? (proc->extension_enabled_const('U') ? MSTATUS_UBE : 0)
```

Comment:
b570b4493768ca6f563b08e7a0c2f7c90fcb6739 looks like 4 separate commits to me

Reply:
Yeah. It contains four point of u/s related check.